src/code.cpp

```diff
@@ -94,9 +100,22 @@ Expr *read_case() {
   }
   // default case
   case 'd': {
-    delete eat_str("efault");
-  }
+    our_ungetc('d');
+    string * already_read = eat_str("default");
```

Comment:
Why not just delete this case and modify `read_ctor()` to recognize `default` as a special case (it looks like it should return `nullptr` in that case)? Then we could just use let the `default:` case below handle this.

Reply:
Good spot! Agreed.

Reply:
Cool, do we still need the larger buffer?

Reply:
I was on the fence about this.

I kept it around because I've encountered a few "buffer full" errors while working with LFSC, but I didn't save the programs that produced them. There's no reason to put both changes into the same PR though -- I'll withdraw the buffer changes from this PR and then create a new PR if I stumble upon a program that triggers this bug again.

Or even better I might change the character stream interface from a get/unget interface to a peek/pop one, since I think we only need 1 character lookahead.